### ISSUE TITLE
0.19 load(), importPromises, undefined System.import() resolved value, and an easy solution

### ISSUE THREAD

**Issue by sberney** [systemjs/systemjs#1628]
For 0.19.45 -- 

Sometimes, intermittently, `System.import('something')` will resolve to undefined, even when `System.get(System.decanonicalize('something'))` will result in the expected ES6 module. I think this really belongs in the es6-module-loader project, but I'm not certain when the change occurred in that project, only when it occurred in system.src.js.

The source of the issue is that `System.load()` is being called before `System.import()`. `System.load()` creates an importPromise, but that specific import promise always resolves to undefined. Previously, it resolved to `evaluateLoadedModule(loader, load)`, aka `load.module.module`.

This appears to be the change, from last year:

https://github.com/systemjs/systemjs/commit/2b8c084f4ad2e5846aeb6157c63e2bc219722542#diff-f64cf64b83a0176d3c8ac2cc0963e63aL819

This is pretty easy to fix, `asyncStartLoadPartwayThrough()` return a promise for the `load`, so about [the line in question](https://github.com/systemjs/systemjs/blob/0.19.45/lib/loader-wrapper-start.js#L792) you can write

```js
      .then(function(load) {
        delete loader.importPromises[name];
        return load.module.module;
      }));
```
Problem solved for me! It's possible this is related to [another recent issue](https://github.com/systemjs/systemjs/issues/1626), but the breaking change appears to have been in there quite some time. (I only recently upgraded.)

**Comment by guybedford** [comment]
Thanks, I've added a fix along these lines and released in 0.19.47.

I don't think this is related to #1626 though unfortunately.

**Comment by sberney** [comment]
Thanks a lot, is there any chance of a jspm-cli version bump sometime soon?
